### PR TITLE
Add request param type information into WebActionMetadata for Wire messages

### PR DESCRIPTION
**Pull request by wesleyk**

### CODE REVIEW

File: misk/src/main/kotlin/misk/web/RequestTypes.kt [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+  private fun handleField(
+      fieldType: KType,
+      fieldName: String,
+      fields: MutableList<Field>,
+      stack: LinkedList<KClass<*>>,
+      repeated: Boolean = false) {
+    val fieldClass = fieldType.classifier as KClass<*>
+    when (fieldClass) {
+      String::class -> fields.add(Field(fieldName, String::class.simpleName!!, repeated))
```
Comment (adrw) [inline misk/src/main/kotlin/misk/web/RequestTypes.kt]: Good central location to find how each fieldClass is processed.

File: misk/src/test/kotlin/misk/ActionsTest.kt [hunk 2]
```diff
@@ -1,36 +1,48 @@
...
-  fun methodReferenceNotAllowedAsAction() {
+  @Test fun noRequestBodyAction() {
+    val action = NoRequestBodyAction::myActionMethod.asAction()
+    assertThat(action.name).isEqualTo("NoRequestBodyAction")
+    assertThat(action.parameterTypes).hasSize(1)
+    assertThat(action.parameterTypes[0]).isEqualTo(Int::class.createType())
+    assertThat(action.requestType).isNull()
+  }
+
+  @Test fun methodReferenceNotAllowedAsAction() {
```
Comment (adrw) [inline misk/src/test/kotlin/misk/ActionsTest.kt]: Could you add a single sentence what this is testing?

File: misk/src/test/kotlin/misk/ActionsTest.kt [hunk 3]
```diff
@@ -1,36 +1,48 @@
...
     assertFailsWith<IllegalArgumentException> {
       val t = TestAction()
       t::myActionMethod.asAction()
     }
   }
 
-  @Test
-  fun freeStandingFunctionNotAllowedAsAction() {
+  @Test fun freeStandingFunctionNotAllowedAsAction() {
```
Comment (adrw) [inline misk/src/test/kotlin/misk/ActionsTest.kt]: Documentation here as well?

File: misk/src/test/kotlin/misk/ActionsTest.kt (hunk 2 above)
Comment (wesleyk) [reply misk/src/test/kotlin/misk/ActionsTest.kt]: these were refactored so the annotation is on the same line as the test. I'm more a fan of the test method name explaining what it's doing. I think the test name is explanatory enough?

**Pull request merged**